server/plugin/client.go

```diff
@@ -100,7 +104,7 @@ func (c *client) CreateTask(body *serializers.CreateTaskRequestPayload, mattermo
 	}
 
 	var task *serializers.TaskValue
-	_, statusCode, err := c.CallPatchJSON(c.plugin.getConfiguration().AzureDevopsAPIBaseURL, taskURL, http.MethodPost, mattermostUserID, &payload, &task, nil)
+	_, statusCode, err := c.CallPatchJSON(c.plugin.getConfiguration().AzureDevopsAPIBaseURL, sanitizedPath, http.MethodPost, mattermostUserID, &payload, &task, nil)
```

Comment:
(suggestion) since the same operation is repeated in multiple places, and all `Call...Json` use client's `Call` method, I suggest performing any transformations for path in `parsePath` that's used by `Call` and also directly by `makeHTTPRequestWithAccessToken` which doesn't use `Call`